Video

bago2k4 opened the pull request:
Add video posts using Ziggeo sdk and out developed theme of betajs-media-components used to display the video stuff.

To test:
- click New
- enter something in the body and leave the headline blank
- [x] do you see an error message if you try to post w/o a headline? Good
- now enter a headline and click post
- [x] did it post? Good
- click New again
- click video button and start recording
- in the video talk a bit to test the transcription later
- while recording add a headline
- while still recording click post
- [x] do you see an error message related to recording still running? Good
- now stop the recording
- before picking a cover shot click post again
- [x] do you see an error message related to the missing cover image? Good
- select a cover image
- wait for the video to finish upload and processing
- [x] do you see only one notification at the end of the upload process saying that it finished uploading and processing? Good
- click Post
- [x] do you see your video in the stream view? Can you play it? Good
- [x] do you see the transcription? Good
- click New post
- add a video a headline and a body
- click Post before finishing uploading
- [x] do you see notifications of the upload progress and of the processing progress? Good
- [x] do you see the video appear in the stream view once the upload is complete? Good
- now add a text post and post it
- edit it
- add a video and save
- go back to the stream view
- [x] do you see the old post with the video and the transcription added? Good
- now edit one of the posts with a video
- remove the video and save
- [x] do you see the post w/o video and transcription in the steam? Good
- switch to grid view
- click on a post with video
- do you see the video in the fullscreen view? Good
- now go back to grid view
- share a post with video and transcription via url with all the world (not only team members)
- open the copied url in a incognito window
- [x] do you see the video player and the transcription? Good


NB: production server is using the Ziggeo sdk compiled but not loaded from the CDN, it's loaded directly from the web server. This should change, we should concat the ziggeo sdk js to oc_assets.js.
It's not a big issue only a bit slower, adding it ot oc_assets should be tested to make sure externs work.

NB: if we decide to remove transcription we need to open a PR and merge also this branch of infra https://github.com/belucid/open-company-infrastructure/tree/fix-video

thepug commented on resources/public/js/externs.js:
```diff
@@ -134,3 +134,22 @@ CarrotGA.NULL_VALUE = {};
+ZiggeoApi.V2.Recorder.activate = function(){};
+ZiggeoApi.V2.Recorder.destroy = function(){};
+ZiggeoApi.V2.Recorder.get = function(){};
+ZiggeoApi.V2 = {};
```
Doesn't this overwrite the previous definitions?

bago2k4 replied:
Good catch, i fixed it but i don't think it was a problem since it was working.

thepug commented on src/oc/web/components/entry_edit.cljs:
```diff
@@ -382,17 +528,25 @@
+                                                     (reset! (::uploading-media s) is-uploading?))
+                               :media-config ["photo" "video"]
+                               :classes "emoji-autocomplete emojiable fs-hide"})]
+          ; (when (and (:fixed-video-id entry-editing)
```
Can this commented out code just be removed?

thepug commented on src/oc/web/components/fullscreen_post.cljs:
```diff
@@ -112,6 +114,7 @@
   (remove-autosave state)
   (reset! (::autosave-timer state) (utils/every 5000 #(autosave state)))
   (.click (js/$ "div.rich-body-editor a") #(.stopPropagation %))
+  ; (ui-utils/resize-textarea (rum/ref-node state "transcript-edit"))
```
also remove

thepug commented on src/oc/web/components/fullscreen_post.cljs:
```diff
@@ -135,21 +138,26 @@
       (dis/dispatch! [:update [:modal-editing-data] #(merge % {:body (utils/clean-body-html raw-html)
-                                                               :has-changes true})]))))
+                                                               :has-changes true})])))
+  ; (let [editing-data (:modal-editing-data @(drv/get-ref state :fullscreen-post-data))]
```
also remove

thepug commented on src/oc/web/components/fullscreen_post.cljs:
```diff
@@ -193,6 +201,61 @@
         (reset! (::initial-headline s) initial-headline)
         (reset! (::edited-data-loaded s) true)))))
 
+(defn remove-video []
```
I'm not really certain why there is duplicated here and https://github.com/open-company/open-company-web/pull/587/files#diff-c257f209ba0537d00a3b216f0e4d992aR150

I can't seem to find where this is used in full screen.

thepug commented on src/oc/web/components/fullscreen_post.cljs:
```diff
@@ -444,9 +547,22 @@
                  :ref :fullscreen-post-box-content-body
-                 :dangerouslySetInnerHTML (utils/emojify (:body activity-data))}])
+                 :dangerouslySetInnerHTML (utils/emojify (:body current-activity-data))}])
+            ; (when (and (:video-transcript current-activity-data)
```
Add a comment since this is related to transcript data we want to use.  Or maybe a feature flag?

thepug commented on src/oc/web/components/secure_activity.cljs:
```diff
@@ -93,19 +108,29 @@
             (when (:body activity-data)
               [:div.activity-body.fs-hide
                 {:dangerouslySetInnerHTML (utils/emojify (:body activity-data))}])
+            ; (when (and (:video-transcript activity-data)
```
Same here, either add a `TODO` comment or feature flag for transcripts.

thepug commented on src/oc/web/components/stream_item.cljs:
```diff
@@ -151,33 +174,51 @@
+                   :class (utils/class-set {:wrt-truncated truncated?
+                                            :wrt-expanded expanded?})
+                   :dangerouslySetInnerHTML (utils/emojify (:body activity-data))}]]]]
+          ; (when (and expanded?
```
more transcript stuff

bago2k4 commented:
@thepug good catches.
I have moved all the commented code under a feature flag from `oc.web.local_settings`.
That require a new PR in infrastructure though:
https://github.com/belucid/open-company-infrastructure/pull/77

to make sure we don't get an error when looking for it on staging or beta.

Also i have moved the `remove-video` duplicated function to the activity actions namespace.
